File: hydra-lisp/hydra-clojure/src/main/clojure/hydra/lib/chars.clj
(ns hydra.lib.chars)

;; is_alpha_num :: Char -> Bool
(def hydra_lib_chars_is_alpha_num
  "Check whether a character is alphanumeric."
  (fn [c] (or (Character/isLetterOrDigit c))))

;; is_lower :: Char -> Bool
(def hydra_lib_chars_is_lower
  "Check whether a character is lowercase."
  (fn [c] (Character/isLowerCase c)))

;; is_space :: Char -> Bool
(def hydra_lib_chars_is_space
  "Check whether a character is a whitespace character."
  (fn [c] (Character/isWhitespace c)))

;; is_upper :: Char -> Bool
(def hydra_lib_chars_is_upper
  "Check whether a character is uppercase."
  (fn [c] (Character/isUpperCase c)))

;; to_lower :: Char -> Char
(def hydra_lib_chars_to_lower
  "Convert a character to lowercase."
  (fn [c] (Character/toLowerCase c)))

;; to_upper :: Char -> Char
(def hydra_lib_chars_to_upper
  "Convert a character to uppercase."
  (fn [c] (Character/toUpperCase c)))

File: hydra-scala/src/main/scala/hydra/lib/chars.scala
package hydra.lib

object chars:
  /** Check whether a character is alphanumeric. */
  def isAlphaNum(c: Int): Boolean = Character.isLetterOrDigit(c)

  /** Check whether a character is lowercase. */
  def isLower(c: Int): Boolean = Character.isLowerCase(c)

  /** Check whether a character is a whitespace character. */
  def isSpace(c: Int): Boolean = Character.isWhitespace(c)

  /** Check whether a character is uppercase. */
  def isUpper(c: Int): Boolean = Character.isUpperCase(c)

  /** Convert a character to lowercase. */
  def toLower(c: Int): Int = Character.toLowerCase(c)

  /** Convert a character to uppercase. */
  def toUpper(c: Int): Int = Character.toUpperCase(c)
